File: frontend/src/types/inventory.ts
export interface Inventory {

    id?: number;

    product_id: number;

    current_stock: number;

    reserved_stock: number;

    available_stock: number;

    reorder_level: number;

    stock_status: string;
}

export interface InventoryDashboard {

    total_products: number;

    total_inventory_quantity: number;

    low_stock_products: number;

    out_of_stock_products: number;
}

export interface StockAdjustment {

    quantity: number;

    reason: string;

    remarks: string;
}

File: backend/app/schemas/inventory.py
from pydantic import BaseModel


class InventoryCreate(BaseModel):
    product_id: int
    current_stock: int
    reserved_stock: int = 0
    reorder_level: int


class InventoryUpdate(BaseModel):
    current_stock: int
    reserved_stock: int
    reorder_level: int


class InventoryResponse(BaseModel):
    id: int
    product_id: int
    current_stock: int
    reserved_stock: int
    available_stock: int
    reorder_level: int
    stock_status: str

class Config:
        from_attributes = True
class StockAdjustment(BaseModel):
    quantity: int
    reason: str
    remarks: str = ""        